ckanext/string_to_location/location_mapper.py

```diff
@@ -60,50 +58,39 @@ def map_location(self):
         # print(f"    {matches_with_polygons} matches with polygons")
         
 
-    def _matches_to_geojson(self, match_array, properties):
+    def _entities_to_geojson(self, entities_array, properties):
             features = []
-            for match in match_array:
-                geometry = match['entity'].geo_polygon.geometry
+            for entity in entities_array:
+                geometry = entity['entity'].geo_polygon.geometry
                 feature_properties = {}
                 for prop in properties:
-                    feature_properties[prop] = match['row'][prop]
+                    feature_properties[prop] = entity['row'][prop]
                 feature = Feature(properties= feature_properties, geometry=geometry)
                 features.append(feature)
 
             return FeatureCollection(features)
 
-    def _build_matches(self, table, column_name, is_name, source_entity_type, target_entity_type):
-        # Build the matches array
-
-        # Attempt to map this from a CSP code into additional entities
-        sources = []
-        matches = []
+    def _build_entities(self, table, column_name, is_name, source_entity_type):
+        entities = []
         errors = []
         rows = 0
         for index, row in table.iterrows():
             rows += 1
-            # For CSP lookup
             lookup_name = row[column_name]
-            # First, build the ONS entity, so we can pass that around consistently
-            # FIXME: this shouldn't be an if statement
+            
             ons_entity = OnsEntityBuilder.build(lookup_name, source_entity_type, is_name=is_name)
 
-            if not isinstance(ons_entity, NullOnsEntity):
-                sources.append(ons_entity)
-                row[source_entity_type.value + '_geojson'] = ons_entity.geo_polygon
-
-            # Second, convert the entity into a LAD (builds a new object, of course)
-            local_authority_district = OnsCodeMapper(ons_entity, target_entity_type).call()
-
-            if isinstance(local_authority_district, NullOnsEntity):
+            # FIXME: this shouldn't be an if statement
+            if isinstance(ons_entity, NullOnsEntity):
                 error_message = "Row:" + str(index) + ", " + lookup_name + " did not match."
                 errors.append(error_message)
             else:
-                matches.append({
-                    'entity': local_authority_district,
+                row[source_entity_type.value + '_geojson'] = ons_entity.geo_polygon
```

Comment:
I don't think we want to append the geojson to the row as well as having it at the top level.